Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -241,6 +243,61 @@ class Frontend {
       }
     }
 
+    // Load global configuration file
+    // First URLs are created, then they are queried. First match is loaded
+    var configLocations: [URL?] = []
+
+    #if os(Windows)
+      if let localAppData = ProcessInfo.processInfo.environment["LOCALAPPDATA"] {
+        configLocations.append(URL(fileURLWithPath: localAppData))
+      }
+      if let programData = ProcessInfo.processInfo.environment["PROGRAMDATA"] {
+        configLocations.append(URL(fileURLWithPath: programData))
+      }
+    #else
+      if let xdgConfigHome = ProcessInfo.processInfo.environment["XDG_CONFIG_HOME"] {
+        configLocations.append(URL(fileURLWithPath: xdgConfigHome))
+      }
+
+      if let libraryUrl = FileManager.default.urls(
+        for: .applicationSupportDirectory, in: .userDomainMask
+      ).first {
+        configLocations.append(libraryUrl)
+      }
+
+      if let xdgConfigDirs = ProcessInfo.processInfo.environment["XDG_CONFIG_DIRS"] {
+        configLocations += xdgConfigDirs.split(separator: ":").map { xdgConfigDir in
+          URL(fileURLWithPath: String(xdgConfigDir))
+        }
+      }
+
+      if let libraryUrl = FileManager.default.urls(
+        for: .applicationSupportDirectory, in: .systemDomainMask
+      ).first {
+        configLocations.append(libraryUrl)
+      }
+    #endif
+
+    for case var location? in configLocations {
+      if #available(macOS 13.0, iOS 16.0, *) {
+        location.append(components: "swift-format", "config.json")
+      } else {
+        location.appendPathComponent("swift-format", isDirectory: true)
+        location.appendPathComponent("config.json", isDirectory: false)
+      }
+      if FileManager.default.fileExists(atPath: location.path) {
+        do {
+          let configuration = try configurationLoader.configuration(at: location)
+          self.checkForUnrecognizedRules(in: configuration)
```

Comment:
What do we need this check for?

Reply:
The check was implemented to print diagnostics when unknown rules are used (https://github.com/swiftlang/swift-format/pull/612). It is checked every time that we load a configuration file

Reply:
Ah, I see. 👍🏽